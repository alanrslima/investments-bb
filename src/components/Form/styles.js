import { StyleSheet } from 'react-native';
import { colors, fonts } from '../../styles';

export default StyleSheet.create({
  container: {
    marginBottom: 15,
    backgroundColor: colors.WHITE,
  },
  containerInput: {
    paddingVertical: 15,
    paddingHorizontal: 20,
  },
  titleInput: {
    fontSize: 13,
    color: colors.BLACK,
    marginBottom: 8,
  },
  input: {
    fontSize: fonts.TITLE,
    color: colors.BLACK,
    fontWeight: '600',
  },
  error: {
    color: colors.RED,
    fontSize: 12,
    marginTop: 5,
  }
});
