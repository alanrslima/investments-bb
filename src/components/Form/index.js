import React, { useState, useEffect } from 'react';
import { View, Text } from 'react-native';
import { TextInputMask } from 'react-native-masked-text';
import PropTypes from 'prop-types';
import Tab from '../Tab';
import styles from './styles';


function Form({ action, onChange }) {
  const [input, setInput] = useState('');
  const [error, setError] = useState('');
  let inputRef = null;

  useEffect(() => {
    if (inputRef) {
      const value = inputRef.getRawValue();
      if (value > action.value) {
        setError(`Valor não pode ser maior que ${action.formatedValue}`);
      } else {
        setError('');
      }
      onChange({ ...action, inputValue: value });
    }
  }, [input]);

  return (
    <View style={styles.container}>
      <Tab
        title="Ação"
        description={action && action.nome}
      />
      <Tab
        title="Saldo acumulado"
        description={action && action.formatedValue}
      />
      <View style={styles.containerInput}>
        <Text style={styles.titleInput}>Valor a resgatar</Text>
        <TextInputMask
          ref={(ref) => { inputRef = ref; }}
          value={input}
          type="money"
          style={styles.input}
          onChangeText={(text) => setInput(text)}
          placeholder="Digite o valor"
          placeholderTextColor="#CCC"
        />
        {error ? <Text style={styles.error}>{error}</Text> : null}
      </View>
    </View>
  );
}

Form.propTypes = {
  action: PropTypes.shape({
    id: PropTypes.string,
    formatedValue: PropTypes.string,
    nome: PropTypes.string,
    value: PropTypes.number,
    percentual: PropTypes.number
  }).isRequired,
  onChange: PropTypes.func.isRequired,
};

export default Form;
