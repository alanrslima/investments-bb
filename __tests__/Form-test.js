import React from 'react';
import renderer from 'react-test-renderer';
import Form from '../src/components/Form';


test('renders correctly', () => {
  const tree = renderer.create(<Form
    action={{
      id: '1',
      nome: 'BBAS3',
      percentual: 41.1,
      formatedValue: 'R$ 10.320,00',
      value: 10320.00
    }}
    onChange={() => { }}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
