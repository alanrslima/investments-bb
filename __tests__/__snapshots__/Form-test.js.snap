// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "marginBottom": 15,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#F4F4F4",
          "borderBottomWidth": 2,
          "marginLeft": 20,
          "paddingRight": 20,
          "paddingVertical": 15,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Ação
        </Text>
        <Text
          style={
            Object {
              "color": "#707070",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          BBAS3
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {},
      ]
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": "#F4F4F4",
          "borderBottomWidth": 2,
          "marginLeft": 20,
          "paddingRight": 20,
          "paddingVertical": 15,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Saldo acumulado
        </Text>
        <Text
          style={
            Object {
              "color": "#707070",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          R$ 10.320,00
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 20,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontSize": 13,
          "marginBottom": 8,
        }
      }
    >
      Valor a resgatar
    </Text>
    <TextInput
      allowFontScaling={true}
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Digite o valor"
      placeholderTextColor="#CCC"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "#000000",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
      type="money"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
